### PR TITLE
Eligible couple tracking bugs done

### PR DESCRIPTION
**Pull request by ritik-bizbrolly**

### CODE REVIEW

File: app/src/main/java/org/piramalswasthya/sakhi/configuration/Dataset.kt [hunk 1]
```diff
@@ -247,6 +247,31 @@ abstract class Dataset(context: Context, currentLanguage: Languages) {
...
 
+
+    protected fun triggerforHide(
+        source: FormElement,
+        passedIndex: Int,
+        triggerIndex: Int,
+        target: FormElement,
+        targetSideEffect: List<FormElement>? = null
+    ): Int {
+
```
Comment (drtechie) [inline app/src/main/java/org/piramalswasthya/sakhi/configuration/Dataset.kt:258]: Indentation is off

File: app/src/main/java/org/piramalswasthya/sakhi/configuration/EligibleCoupleTrackingDataset.kt [hunk 2]
```diff
@@ -112,7 +114,7 @@ class EligibleCoupleTrackingDataset(
             financialYear,
             month,
             isPregnancyTestDone,
-            isPregnant,
+//            isPregnant,
```
Comment (drtechie) [inline app/src/main/java/org/piramalswasthya/sakhi/configuration/EligibleCoupleTrackingDataset.kt:117]: remove comments

File: app/src/main/java/org/piramalswasthya/sakhi/configuration/EligibleCoupleTrackingDataset.kt [hunk 3]
```diff
@@ -190,37 +195,147 @@ class EligibleCoupleTrackingDataset(
...
                 isPregnant.isEnabled = true
-                triggerDependants(
-                    source = isPregnancyTestDone,
-                    passedIndex = index,
-                    triggerIndex = 0,
-                    target = pregnancyTestResult
-                )
+
+                if (isPregnant.value==resources.getStringArray(R.array.yes_no_donno)[0])
+                {
```
Comment (drtechie) [inline app/src/main/java/org/piramalswasthya/sakhi/configuration/EligibleCoupleTrackingDataset.kt:200]: Follow same pattern with curly braces? Some places it's in new line, some places it's with the if condition.

File: app/src/main/java/org/piramalswasthya/sakhi/configuration/EligibleCoupleTrackingDataset.kt [hunk 4]
```diff
@@ -190,37 +195,147 @@ class EligibleCoupleTrackingDataset(
...
             }
 
             pregnancyTestResult.id -> {
+
+
                 if (pregnancyTestResult.value == resources.getStringArray(R.array.ectdset_po_neg)[0]) {
                     isPregnant.value = resources.getStringArray(R.array.yes_no)[0]
                     isPregnant.isEnabled = false
-                } else {
+                    Log.e("Positive","$index ${pregnancyTestResult.value}")
```
Comment (drtechie) [inline app/src/main/java/org/piramalswasthya/sakhi/configuration/EligibleCoupleTrackingDataset.kt:252]: remove logs

File: app/src/main/java/org/piramalswasthya/sakhi/configuration/EligibleCoupleTrackingDataset.kt [hunk 5]
```diff
@@ -190,37 +195,147 @@ class EligibleCoupleTrackingDataset(
...
+                        targetSideEffect = listOf(isPregnant,usingFamilyPlanning,methodOfContraception, anyOtherMethod)
+                    )
+                    triggerDependants(
+                        source = isPregnant,
+                        passedIndex = index,
+                        triggerIndex = 1,
+                        target = usingFamilyPlanning,
+                        targetSideEffect = listOf(methodOfContraception, anyOtherMethod)
+                    )
+
```
Comment (drtechie) [inline app/src/main/java/org/piramalswasthya/sakhi/configuration/EligibleCoupleTrackingDataset.kt:287]: remove unnecessary line breaks and logs.

File: app/src/main/java/org/piramalswasthya/sakhi/configuration/PregnantWomanRegistrationDataset.kt [hunk 6]
```diff
@@ -22,11 +23,20 @@ class PregnantWomanRegistrationDataset(
...
         private fun getMinLmpMillis(): Long {
             val cal = Calendar.getInstance()
             cal.add(Calendar.DAY_OF_YEAR, -1 * 280)
             return cal.timeInMillis
         }
+
+        private fun pwRegisterBeforeoneYear(): Long {
+            Log.e("RegstrationDate:-" , registrationDate.toString())
+            return   registrationDate-TimeUnit.DAYS.toMillis(365)
+           /* val calendar = Calendar.getInstance()
```
Comment (drtechie) [inline app/src/main/java/org/piramalswasthya/sakhi/configuration/PregnantWomanRegistrationDataset.kt:36]: logs and commented code to be removed.

File: app/src/main/java/org/piramalswasthya/sakhi/configuration/PregnantWomanRegistrationDataset.kt [hunk 7]
```diff
@@ -160,6 +170,7 @@ class PregnantWomanRegistrationDataset(
         title = resources.getString(R.string.pwrdst_vdrl_tst_done),
         arrayId = -1,
         required = false,
+        min=pwRegisterBeforeoneYear(),
```
Comment (drtechie) [inline app/src/main/java/org/piramalswasthya/sakhi/configuration/PregnantWomanRegistrationDataset.kt:173]: space around =

**Review by drtechie — CHANGES_REQUESTED** [review]
Several clean code comments.

**Comment by drtechie** [comment]
Add a description of what is addressed here.

**Comment by ritik-bizbrolly** [comment]
Hello,

I have made the required changes and will follow the guidelines from here on.


Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Review by sushant-bizbrolly — COMMENTED** [review]
Changes are done.

**Review by drtechie — APPROVED** [review]

**Pull request merged**